lib/internal/process/execution.js

```diff
@@ -33,6 +33,24 @@ function tryGetCwd() {
   }
 }
 
+function evalModule(source) {
+  const { decorateErrorStack } = require('internal/util');
+  const asyncESM = require('internal/process/esm_loader');
+  asyncESM.loaderPromise.then(async (loader) => {
```

Comment:
Non-blocking: Is it intentional that this does not use `async / await` but plain promises? I would rather use `async / await`.

Reply:
There is an async function internally here, but note that the outer promise is not propagated.

Reply:
I saw that the promise is not returned, it just seems cleaner to me if `evalModule` would be an async function as well so that it's possible to await `asyncESM.loaderPromise` and to use `try / catch` in there. Especially with the new async stack traces that only work when using `async / await`.

Reply:
However it is done, we still need something to do the outer `.catch` handler, at least until we have an async bootstrap call from the C++ side (I did look into that in the early days of modules, but that broke a lot of things...!).

So the only `.then` we can save is the `loaderPromise` one but that would likely be at the cost of another wrapper I think.